integration/presto/src/main/prestodb/org/apache/carbondata/presto/CarbondataSplitManager.java

```diff
@@ -109,8 +109,15 @@ public ConnectorSplitSource getSplits(ConnectorTransactionHandle transactionHand
     if (!table.getStorage().getStorageFormat().getInputFormat().contains("carbon")) {
       return super.getSplits(transactionHandle, session, layoutHandle, splitSchedulingStrategy);
     }
-    String location = table.getStorage().getLocation();
-
+    String location;
+    String tablePath = table.getStorage().getSerdeParameters().get("tablePath");
```

Comment:
it is better like 
```
String location = table.getStorage().getSerdeParameters().get("tablePath");
if (StringUtils.isEmpty(location))  {
   location = table.getStorage().getLocation();
}
```

Reply:
I think you mean StringUtils.isEmpty(`tablePath`) not `location`.
But yeah, got your intention. I will change

Reply:
got it. ok.

Reply:
done